services/src/main/scala/cromwell/services/metadata/impl/archiver/ArchiveMetadataSchedulerActor.scala

```diff
@@ -130,48 +132,57 @@ class ArchiveMetadataSchedulerActor(archiveMetadataConfig: ArchiveMetadataConfig
     })
   }
 
-  def archiveNextWorkflow(startTime: OffsetDateTime): Future[Boolean] = {
+  def archiveNextWorkflows(startTime: OffsetDateTime): Future[Long] = {
     for {
-      maybeWorkflowSummaryEntry <- lookupNextWorkflowToArchive()
-
+      workflowSummaryEntries <- lookupNextWorkflowsToArchive(archiveMetadataConfig.batchSize)
       workflowSelectedTime = OffsetDateTime.now()
-      _ = sendTiming(archiverTimingMetricsBasePath :+ "lookup_next_workflow", calculateTimeDifference(startTime, workflowSelectedTime), ServicesPrefix)
-
-      result <- maybeWorkflowSummaryEntry match {
-        case Some(summaryEntry) =>
-          summaryEntry.endTimestamp.foreach { workflowEndTime =>
-            val millisSinceWorkflowEnd = JDuration.between(workflowEndTime.toSystemOffsetDateTime, OffsetDateTime.now()).toMillis
-            sendGauge(timeBehindExpectedDelayMetricPath, millisSinceWorkflowEnd - archiveMetadataConfig.archiveDelay.toMillis, ServicesPrefix)
-          }
-          for {
-            path <- Future.fromTry(getGcsPathForMetadata(summaryEntry))
-            workflowId = summaryEntry.workflowExecutionUuid
-            dbStream <- fetchStreamFromDatabase(WorkflowId(UUID.fromString(workflowId)))
-            _ = log.info(s"Archiving metadata for $workflowId to ${path.pathAsString}")
-            readyToStreamTime = OffsetDateTime.now()
-            _ = sendTiming(archiverTimingMetricsBasePath :+ "prepare_to_stream", calculateTimeDifference(workflowSelectedTime, readyToStreamTime), ServicesPrefix)
-            _ <- streamMetadataToGcs(path, dbStream)
-            streamCompleteTime = OffsetDateTime.now()
-            _ = sendTiming(archiverTimingMetricsBasePath :+ "stream_to_gcs", calculateTimeDifference(readyToStreamTime, streamCompleteTime), ServicesPrefix)
-            _ <- updateMetadataArchiveStatus(WorkflowId(UUID.fromString(workflowId)), Archived)
-            statusUpdatedTime = OffsetDateTime.now()
-            _ = sendTiming(archiverTimingMetricsBasePath :+ "archive_status_update", calculateTimeDifference(streamCompleteTime, statusUpdatedTime), ServicesPrefix)
-            _ = log.info(s"Archiving succeeded for $workflowId")
-          } yield true
-        case None =>
-          sendGauge(timeBehindExpectedDelayMetricPath, 0L, ServicesPrefix)
-          Future.successful(false)
-      }
+      _ = sendTiming(archiverTimingMetricsBasePath :+ "lookup_next_workflows", calculateTimeDifference(startTime, workflowSelectedTime), ServicesPrefix)
```

Comment:
I think the graphs in grafana already use a `*` to cover all the timing values, but consider this a request to check that both the old _AND_ new value can be displayed at the same time, so that we don't need to remember to update them when the new Cromwell version is published.

Reply:
Checked and confirmed working. Screenshot from a window where both current release & my custom JAR were running.
![Screenshot_2021-05-13 Cromwell Archiver - Grafana(5)](https://user-images.githubusercontent.com/1087943/118197462-01ab8000-b41d-11eb-86ab-b2f7307cb294.png)